src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -328,12 +328,22 @@ protected string GenerateCommandLineCommands (string ManifestFile, string curren
 				cmd.AppendSwitchIfNotNull ("-F ", currentResourceOutputFile + ".bk");
 			// The order of -S arguments is *important*, always make sure this one comes FIRST
 			cmd.AppendSwitchIfNotNull ("-S ", resourceDirectory.TrimEnd ('\\'));
-			if (AdditionalResourceDirectories != null)
-				foreach (var resdir in AdditionalResourceDirectories)
-					cmd.AppendSwitchIfNotNull ("-S ", resdir.ItemSpec.TrimEnd ('\\'));
-			if (AdditionalAndroidResourcePaths != null)
-				foreach (var dir in AdditionalAndroidResourcePaths)
-					cmd.AppendSwitchIfNotNull ("-S ", Path.Combine (dir.ItemSpec.TrimEnd (System.IO.Path.DirectorySeparatorChar), "res"));
+			if (AdditionalResourceDirectories != null) {
```

Comment:
Do we really need to repeat this much code, changing only the `ITaskItem[]` source?

For example, with C# local functions:

```csharp
void AddResourceDirectories(ITaskItem[] directories) {
    if (directories == null) {
        return;
    }
    foreach (var dir in directories) {
        if (!Directory.Exists (dir.ItemSpec))
            continue;
        var resdir = dir.ItemSpec.TrimEnd ('\\');
        cmd.AppendSwitchIfNotNull ("-S ", resdir);
    }
}
AddResourceDirectories (AdditionalResourceDirectories);
AddResourceDirectories (AdditionalAndroidResourcePaths);
```

Reply:
One does a `Path.Combine (dir, "res")` and the other does `TrimEnd()`.

I could use a local function, here but I'd also have to do something like:
```csharp
void AddResourceDirectories(ITaskItem[] directories, Func<string, string> func) {
    if (directories == null) {
        return;
    }
    foreach (var dir in directories) {
        if (!Directory.Exists (dir.ItemSpec))
            continue;
        var resdir = func (dir.ItemSpec);
        cmd.AppendSwitchIfNotNull ("-S ", resdir);
    }
}
AddResourceDirectories (AdditionalResourceDirectories, d => d.TrimEnd ('\\'));
AddResourceDirectories (AdditionalAndroidResourcePaths, d => Path.Combine (d, "res"));
```
At that point, not sure if it's simpler?